### PR TITLE
tests: Add tests for TypedArray#sort

### PR DESCRIPTION
**Pull request by saulecabrera**
Adds missing tests for `TypedArray#sort`

- [x] I've read the contributing guidelines

### CODE REVIEW

File: tests/compiler/std/typedarray.ts [hunk 1]
```diff
@@ -845,5 +845,45 @@ testTypedArraySet<Float64Array>();
...
+  source[2] = <T>1;
+  source[1] = <T>2;
+  source[0] = <T>3;
+
+  let result = source.sort();
+  assert(result[0] == <T>1);
+  assert(result[1] == <T>2);
+  assert(result[2] == <T>3);
+
+  result = result.sort(function(a: T, b: T) {
```
Comment (saulecabrera) [inline tests/compiler/std/typedarray.ts:859]: Could've done `b - a` but I didn't want to cast depending on the type of `T`.

**Review by MaxGraey — APPROVED** [review]

File: tests/compiler/std/typedarray.ts (hunk 1 above)
Comment (MaxGraey) [reply tests/compiler/std/typedarray.ts:859]: You could just use 
```ts
result.sort();
```
instead:
```ts
  result = result.sort(function(a: T, b: T) { ... });
```

**Comment by saulecabrera** [comment]
@dcodeIO this one is ready to go.

**Comment by dcodeIO** [comment]
Thanks! :)

**Pull request merged**